cirq/google/calibration/workflow.py

```diff
@@ -300,24 +303,36 @@ def run_characterizations(
     if not calibrations:
         return []
 
-    requests = [
-        [
-            calibration.to_calibration_layer()
-            for calibration in calibrations[offset : offset + max_layers_per_request]
-        ]
-        for offset in range(0, len(calibrations), max_layers_per_request)
-    ]
+    if isinstance(engine, Engine):
+        if processor_id is None:
+            raise ValueError('processor_id must be provided when running on the engine')
+        if gate_set is None:
+            raise ValueError('gate_set must be provided when running on the engine')
 
-    results = []
+        results = []
 
-    for request in requests:
-        job = engine.run_calibration(request, processor_id=processor_id, gate_set=gate_set)
-        results += [
-            calibration.parse_result(result)
-            for calibration, result in zip(calibrations, job.calibration_results())
+        requests = [
+            [
+                calibration.to_calibration_layer()
+                for calibration in calibrations[offset : offset + max_layers_per_request]
+            ]
+            for offset in range(0, len(calibrations), max_layers_per_request)
         ]
-        if progress_func:
-            progress_func(len(results), len(calibrations))
+
+        for request in requests:
+            job = engine.run_calibration(request, processor_id=processor_id, gate_set=gate_set)
+            request_results = job.calibration_results()
+            results += [
+                calibration.parse_result(result)
+                for calibration, result in zip(calibrations, request_results)
+            ]
+            if progress_func:
+                progress_func(len(results), len(calibrations))
+
+    elif isinstance(engine, PhasedFSimEngineSimulator):
+        results = engine.get_calibrations(calibrations)
```

Comment:
this massive if/else around the whole function body with different argument types is a code smell. Can we add a `run_calibration` mock to `PhasedFSimEngineSimulator`?

Reply:
I feel that adding proto/message level support to `PhasedFSimEngineSimulator` is the wrong way around. We should rather have the `q.e.Sampler` that supports the higher level interface instead. I wouldn't like to do it just yet though.

Reply:
Offline discussion with @mrwojtek that this would solve my issues, but we'll do it in a follow-up PR